lib/GoalCohort.js

```diff
@@ -317,6 +317,15 @@ GoalCohort.statics.createTeam = async function(user, goal, teamname) {
   let teamIcon = "LogoTeamRazorfish";
 
   if (cohort) {
+    for (let i = 0; i < cohort.teams.length; i++) {
+      if (cohort.teams[i].name === teamname) {
+        const err = new Error(
+          `Cohort already has a team ${teamname}. Please pick a different name`
+        );
+        err.status = 409;
```

Comment:
This is that weird situation where we know the caller wants an HTTP error type but this module really shouldn't know anything about HTTP. 

Longer term, maybe we should use custom exception types that callers like `pal-api` could then check for, but for now let's use the [http-errors](https://www.npmjs.com/package/http-errors) npm module, and then this could be:

```js
  throw createError(409, `Cohort already has a team ${teamname}. Please pick a different name`);
```